include/root_finders/quadratic_polynomial_solvers.hpp

```diff
@@ -0,0 +1,77 @@
+// VolEsti (volume computation and sampling library)
+// Copyright (c) 2021 Vissarion Fisikopoulos
+// Copyright (c) 2021 Apostolos Chalkis
+
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+
+#ifndef QUADRATIC_POLYNOMIAL_SOLVERS_H
+#define QUADRATIC_POLYNOMIAL_SOLVERS_H
+
+// The function returns the sigh of the input number
+template <typename T> int sgn(T val) 
+{
+    return (T(0) < val) - (val < T(0));
+}
+
+// The function compute the roots of a quadratic polynomial equation
+template <typename NT>
+void solve_quadratic_polynomial(NT a, NT b, NT c, NT &x1, NT &x2, bool &real) {
```

Comment:
I think you should also add the case when `a == 0`.

Reply:
the parameters `a, b, c` could be `const&`

Reply:
Thanks for the comment. Done.